libraries/AP_L1_Control/AP_L1_Control.cpp

```diff
@@ -79,7 +79,14 @@ int32_t AP_L1_Control::get_yaw_sensor() const
 int32_t AP_L1_Control::nav_roll_cd(void) const
 {
     float ret;
-    ret = cosf(_ahrs.pitch)*degrees(atanf(_latAccDem * (1.0f/GRAVITY_MSS)) * 100.0f);
+	/*
+		formula can be obtained through equations of balanced spiral:
+		liftForce * cos(roll) = gravityForce * cos(pitch);
+		liftForce * sin(roll) = gravityForce * lateralAcceleration / gravityAcceleration; // as mass = gravityForce/gravityAcceleration
+		see issue 24319 [https://github.com/ArduPilot/ardupilot/issues/24319]
+		Multiplier 100.0f is for converting degrees to centidegrees 
+	*/ 
+    ret = degrees(atanf(_latAccDem * (1.0f/(GRAVITY_MSS * cosf(_ahrs.pitch))))) * 100.0f;
```

Comment:
thanks for this, we just need to protect against cosf(pitch) being zero (think aerobatics)

Reply:
Thank you for comment. To not fail the calculations in case of `_ahrs.pitch ` (`if(fabs(fabs(ahrs.pitch) - 1.5707964f) <= FLT_EPSILON)`)is close to ±90 degrees, `atan2f` can be used in the formula or the previous roll  `_ahrs.roll`can be kept until pitch changes from ±90 degrees. But several checks also should be done to choose correct strategy.

Reply:
I'd be happy to just use constrain_float() to keep the pitch used in the calculation between +/- 60 degrees. The L1 controller is not likely to be useful beyond this anyway. The point of the check is to prevent a numerical error if you are doing aerobatics when you switch into a L1 nav mode (eg. RTL failsafe)

Reply:
Should I update the branch with rebase and add your suggestion?

Reply:
Dear @tridge the changes are made and approved by @priseborough. Do we need more updates here?

Reply:
This has been fixed